File: javascript/0104. Maximum Depth of Binary Tree.js
// INPUT: 3
//       / \
//      9   20
//         /  \
//        15   7

const maxDepth = (node) => {
  if (node === null) {
    return 0;
  }

  return 1 + Math.max(maxDepth(node.left), maxDepth(node.right));
};

console.log(
  maxDepth({
    val: 3,
    right: {
      val: 20,
      right: { val: 7, right: null, left: null },
      left: { val: 15, right: null, left: null },
    },
    left: { val: 9, right: null, left: null },
  })
);

File: python/0104. Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxDepth(self, root: TreeNode, depth: int = 0) -> int:
        if root is None:
            return depth

        depth += 1

        return max(self.maxDepth(root.left, depth), self.maxDepth(root.right, depth))
